packages/polaris-test/tests/mandatory-headers.test.ts

```diff
@@ -0,0 +1,80 @@
+import { graphQLRequest } from '../test-utils/graphql-client';
+import { createServers } from '../test-utils/tests-servers-util';
+import * as allBooks from './jsonRequestsAndHeaders/allBooks.json';
+import * as introspectionQuery from './jsonRequestsAndHeaders/introspectionQuery.json';
+import axios from 'axios';
+import * as polarisProperties from '../shared-resources/polaris-properties.json';
+
+describe('mandatory headers', () => {
+    test.each(createServers({ allowMandatoryHeaders: true }))(
+        'not requested in introspection query',
+        async (server) => {
+            await server.start();
+            const url = `http://localhost:${polarisProperties.port}/${polarisProperties.version}/graphql`;
+            const result = await axios.post(url, {
+                query: introspectionQuery.request,
+                operationName: 'IntrospectionQuery',
+            });
+            expect(result.status).toEqual(200);
+            await server.stop();
+        },
+    );
+    test.each(createServers({ allowMandatoryHeaders: true }))(
+        'both headers are missing',
+        async (server) => {
+            await server.start();
+            expect.assertions(1);
+            try {
+                await graphQLRequest(allBooks.request);
```

Comment:
if you're trying to test some async logic, make sure you test it like so:

`await expect(action).rejects.toEqual(new Error('Something wrong'));`